### ISSUE TITLE
@types/react-redux broken on new type definition

### ISSUE THREAD

**Issue by horiuchi** [DefinitelyTyped/DefinitelyTyped#18955]
- [x] I tried using the `@types/react-redux` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @tkqubo @seansfkelley @thasner @kenzierocks @clayne11

I hope to success to compile below code. But `<Header />` line filed to compile.
```
import * as React from 'react';
import { ActionCreator } from 'redux';
import { connect } from 'react-redux';

const onClick: ActionCreator<{}> = null;
const mapStateToProps = (state: any) => {
    return {
        title: state.app.title as string,
    };
};
const dispatchToProps = {
    onClick,
};

type Props = { title: string; } & typeof dispatchToProps;
const HeaderComponent: React.StatelessComponent<Props> = (props) => {
    return <h1>{props.title}</h1>;
}

const Header = connect(mapStateToProps, dispatchToProps)(HeaderComponent);
<Header />
```

**Comment by begincalendar** [comment]
v5.0.2 doesn't work properly for me when I try to call `connect()` with no "mapDispatchToProps" function.

Reverting to v5.0.1 fixes the issue.

**Comment by clayne11** [comment]
As far as I know the second argument of `connect` doesn't accept an object, just a function or a function that returns a function. In terms of not not working with no `mapDispatchToProps`, you might be running into the issue that's being addressed in #18984.

**Comment by horiuchi** [comment]
@clayne11  No, that is correct usage. It's written in the official API document.

> [mapDispatchToProps(dispatch, [ownProps]): dispatchProps] (Object or Function): If an object is passed, each function inside it is assumed to be a Redux action creator. An object with the same function names, but with every action creator wrapped into a dispatch call so they may be invoked directly, will be merged into the component’s props.

via. https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options
